daemon/loop.go

```diff
@@ -22,7 +23,8 @@ const (
 // Loop for potentially long-running stuff. This includes running
 // jobs, and looking for new commits.
 
-func (d *Daemon) Loop(stop chan struct{}, logger log.Logger) {
+func (d *Daemon) Loop(stop chan struct{}, wg *sync2.WaitGroup, logger log.Logger) {
+	defer wg.Done()
```

Comment:
Oh! Yes very good point :blush: . That suggests to me that `WaitGroup`s aren't suitable for things that get called as `go Loop(...)`*. But, I think we _do_ want a `WaitGroup` here (so that one can be used for both the queue and the daemon, after removing the shutdown order dependency), so let's stick with it.